pkg/cmd/set_replication.go

```diff
@@ -0,0 +1,231 @@
+	v := viper.New()
+	o := NewFinalizeOptions(streams)
+	cmd := &cobra.Command{
+		Use:     "set-replication [OPTIONS]",
+		Short:   i18n.T("Set and pause a scribe replication and ensure a destination PVC with synced data."),
+		Long:    fmt.Sprint(scribeSetReplicationLong),
+		Example: fmt.Sprint(scribeSetReplicationExample),
+		Version: ScribeVersion,
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete())
+			kcmdutil.CheckErr(o.SetReplication())
+		},
+	}
+	kcmdutil.CheckErr(o.Config.Bind(cmd, v))
+	o.RepOpts.Bind(cmd, v)
+	kcmdutil.CheckErr(o.Bind(cmd, v))
+
+	return cmd
+}
+
+func (o *FinalizeOptions) bindFlags(cmd *cobra.Command, v *viper.Viper) error {
+	flags := cmd.Flags()
+	flags.StringVar(&o.sourceName, "source-replication-name", o.sourceName, "name of ReplicationSource (default '<source-ns>-source')")
+	flags.StringVar(&o.destName, "dest-replication-name", o.destName, "name of ReplicationDestination (default '<dest-ns>-destination') ")
+	flags.StringVar(&o.destPVC, "dest-pvc", o.destPVC, "name of not-yet-existing destination PVC. Default is sourcePVC name, or if PVC with sourcePVC name exists in destination namespace, then 'sourcePVC-<date-tag>'")
+	flags.StringVar(&o.destCapacity, "dest-capacity", o.destCapacity, "size of the destination volume to create. Default is source volume capacity.")
+	flags.StringVar(&o.destStorageClass, "dest-storage-class", o.destStorageClass, "name of the StorageClass of the destination volume. If not set, the default StorageClass will be used.")
+	flags.DurationVar(&o.timeout, "timeout", time.Minute*5, "length of time to wait for final sync to complete. Default is 5m. Pass values as time unit (e.g. 1,, 2m, 3h)")
+	flags.VisitAll(func(f *pflag.Flag) {
+		if !f.Changed && v.IsSet(f.Name) {
+			val := v.Get(f.Name)
+			kcmdutil.CheckErr(flags.Set(f.Name, fmt.Sprintf("%v", val)))
+		}
+	})
+	return nil
+}
+
+func (o *FinalizeOptions) Bind(cmd *cobra.Command, v *viper.Viper) error {
+	v.SetConfigName(scribeConfig)
+	v.AddConfigPath(".")
+	v.SetConfigType("yaml")
+	if err := v.ReadInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
+			return err
+		}
+	}
+	return o.bindFlags(cmd, v)
+}
+
+func (o *FinalizeOptions) Complete() error {
+	if err := o.RepOpts.Complete(); err != nil {
+		return err
+	}
+	if len(o.destName) == 0 {
+		o.destName = fmt.Sprintf("%s-destination", o.RepOpts.Dest.Namespace)
+	}
+	if len(o.sourceName) == 0 {
+		o.sourceName = fmt.Sprintf("%s-source", o.RepOpts.Source.Namespace)
+	}
+	return nil
+}
+
+//nolint:funlen
+// SetReplication does the following:
+// 1) Performs manually triggered sync as the last sync
+// 2) Create DestinationPVC if CopyMethod=Snapshot
+// With the manual trigger in place, no further replications will execute.
+func (o *FinalizeOptions) SetReplication() error {
+	t := time.Now().Format("2006-01-02T15:04:05")
+	tag := strings.ReplaceAll(t, ":", "-")
+	lastManualSync := strings.ToLower(tag)
```

Comment:
I think you can do all the formatting w/ time.Format():
```suggestion
	lastManualSync := time.Now().Format("2006-01-02t15-04-05")
```

Reply:
:facepalm: cool, thanks!

Reply:
updated